src/app/(proper_react)/redesign/(authenticated)/admin/feature-flags/components/AddFeatureFlag.tsx

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { useRouter } from "next/navigation";
+import { FormEvent } from "react";
+
+// TODO investigate way to harmonize this with `FeatureFlag` type
+interface FeatureFlagElement {
+  name: {
+    value: string;
+  };
+  isEnabled: {
+    checked: string;
+  };
+  dependencies: {
+    value: string;
+  };
+  allowList: {
+    value: string;
+  };
+  waitList: {
+    value: string;
+  };
+  owner: {
+    value: string;
+  };
+}
+
+export const AddFeatureFlag = () => {
+  const router = useRouter();
+
+  const handleSubmit = (
+    event: FormEvent<HTMLFormElement> & {
+      target: FeatureFlagElement;
+    }
+  ) => {
+    event.preventDefault();
+
+    const eventTarget: FeatureFlagElement = event.target;
+
+    const data = {
+      name: eventTarget.name.value,
+      isEnabled: eventTarget.isEnabled.checked ? true : false,
+      dependencies: eventTarget.dependencies.value.split(","),
+      allowList: eventTarget.allowList.value.split(","),
+      waitList: eventTarget.waitList.value.split(","),
+      owner: eventTarget.owner.value,
+    };
+
```

Comment:
Might be nice to add a empty check for `data.name`. I accidentally clicked the add button too many times and ended up with a few empty rows

Reply:
Yeah same :(